package com.extfro.extfrocore.api.cover.filter;

import com.lowdragmc.lowdraglib2.gui.ui.UIElement;

import java.util.function.Consumer;
import java.util.function.Predicate;

public interface Filter<T, S extends Filter<T, S>> extends Predicate<T> {

    UIElement openConfigurator(int x, int y);

    void setOnUpdated(Consumer<S> onUpdated);

    default boolean isBlackList() {
        return false;
    }

    default boolean isBlank() {
        return false;
    }
}
